### ISSUE TITLE
Add auto update imports on file rename

### ISSUE THREAD

**Issue by benediktms** [LunarVim/LunarVim#3163]
### Feature Description

Modern IDEs/Text Editors like VSCode have the ability to automatically update import statements where the language server supports this (e.g. for TypeScript and Rust). For large code bases where a import may be used dozens if not hundreds of times this is a real life saver. It would be great to have similar funcitonality in LunarVim where appropriate.

### Describe the alternatives you have considered

Going into a seperate editor like VSCode to facilitate the refactor

**Comment by abzcoding** [comment]
we should already have this ( for example in rust) 
though we might have better support for them (after [these PRs](https://github.com/LunarVim/starter.lvim/pull/16#issuecomment-1269147404) get merged)

for example, hitting in the following picture, selecting Error, and hitting `<CR>` would auto import the proper thing

<img width="1920" alt="Screenshot 2022-10-06 at 6 44 03 PM (2)" src="https://user-images.githubusercontent.com/10992695/194351613-a75b83b5-1b6f-498a-aeb2-8648c2d4e4d0.png">
<img width="1920" alt="Screenshot 2022-10-06 at 6 44 19 PM (2)" src="https://user-images.githubusercontent.com/10992695/194351705-ec0de833-ddae-4957-b75f-d9ec2324b4cc.png">

**Comment by benediktms** [comment]
Yes the auto import works, but I was thinking more about LunarVim detecting changes to the file name itself (following the Rust example, if a module/folder changes name). See pictures
![image](https://user-images.githubusercontent.com/48836135/194355731-b0672faa-5aca-4052-8ca0-954f2703799b.png)

If I change the name of the `routes` folder I would like the module to detect these changes and update accordingly:

![image](https://user-images.githubusercontent.com/48836135/194356849-530ddb2c-4d36-475f-abca-4e39a4e67c4c.png)

**Comment by crushingCodes** [comment]
@benediktms Did you end up finding a workaround?

I wonder if it would it be possible to implement something like this in lunarvim? https://github.com/neovim/neovim/issues/20784#issuecomment-1288085253

Not as good as the suggestion above but at least allows a way to run a command to get the desired result

**Comment by olalonde** [comment]
I believe if you create this file file `lua/lvim/lsp/providers/tsserver.lua` with this content, you should be able to use `:RenameFile` command.

```lua

local function rename_file()
    local source_file, target_file

    vim.ui.input({
        prompt = "Source : ",
        completion = "file",
        default = vim.api.nvim_buf_get_name(0)
    },
        function(input)
            source_file = input
        end
    )
    vim.ui.input({
        prompt = "Target : ",
        completion = "file",
        default = source_file
    },
        function(input)
            target_file = input
        end
    )

    local params = {
        command = "_typescript.applyRenameFile",
        arguments = {
            {
                sourceUri = source_file,
                targetUri = target_file,
            },
        },
        title = ""
    }

    vim.lsp.util.rename(source_file, target_file)
    vim.lsp.buf.execute_command(params)
end


return {
    commands = {
        RenameFile = {
            rename_file,
            description = "Rename File"
        },
    }
}

```

Credit for the function: https://github.com/neovim/neovim/issues/20784#issuecomment-1288085253

**Comment by crushingCodes** [comment]
@olalonde thanks link led me to find [antosha417/nvim-lsp-file-operations](https://github.com/antosha417/nvim-lsp-file-operations) which works for this use case
